test/es-module/test-esm-loader.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+// Flags: --expose-internals
+
+// This test ensures that resolve and search throw errors appropriately
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const { join } = require('path');
+const resolve = require('internal/modules/esm/default_resolve');
+const { writeFileSync, unlinkSync } = require('fs');
+
+const { search } = resolve;
+const unsupported = join(
+  tmpdir.path,
+  'module-with-extension-that-is.unsupported');
+tmpdir.refresh();
+writeFileSync(unsupported, '');
+common.expectsError(
+  () => resolve(unsupported, 'file://' + require.resolve(__filename)),
+  {
+    code: 'ERR_UNKNOWN_FILE_EXTENSION',
+    type: TypeError,
+    message: 'Unknown file extension: ' + unsupported
+  }
+);
```

Comment:
Would you be so kind and post an example how a user could actually run into such an error without using any internals directly?

Reply:
`import 'foo' from 'file://path/to/module.unsupported'`